### DISCUSSION TITLE
Show Services which  aren't web based

### DISCUSSION THREAD

**Discussion by KaratekHD** [upptime/upptime#162] [Q&A]
Is it possible to monitor for example a telegram bot? Or to just make uptime show it is down  by opening an issue?

**Comment by AnandChowdhary** [comment]
Curious, how would you go about monitoring a Telegram bot? Would you sent it a message every few mins and see if it replies with expected text?

**Reply by KaratekHD** [reply]
My idea was to check its uptime using the integrated REST API and open an issue in the Uptime Repo whenever it is down

**Reply by Soneji** [reply]
If it uses a REST API surely you can use upptime as it is? You can just specify the API endpoint as a site URL

**Reply by KaratekHD** [reply]
That would of course be possible, but the REST API is only accessible from inside our Intranet

**Reply by Soneji** [reply]
You could write a proxy for that single endpoint. I’d be happy to help using something like nginx or Apache

**Reply by KaratekHD** [reply]
Thank you for offering your help, but sadly this is not possible either as my father does not allow any incoming traffic to my servers 😔

**Reply by Soneji** [reply]
Maybe you can write a program to check if the telegram bot is alive on your server, and if it it down, you can use the github api to create an issue on your upptime repository. This (correct me if im wrong) will show up on the web dashboard. 

https://docs.github.com/en/free-pro-team@latest/rest/reference/issues#create-an-issue

**Reply by Soneji** [reply]
If your servers do not allow incoming traffic, I do not think any upptime code can help really
